Mathlib/Order/SigmaCompleteLattice.lean

```diff
@@ -0,0 +1,393 @@
+/-
+Copyright (c) 2025 Pierre Quinton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pierre Quinton
+-/
+import Mathlib.Data.Set.Countable
+
+/-!
+# σ-complete lattices
+
+This file contains the definition and some properties of σ-complete lattices.
```

Comment:
Can you explain what such lattices are in the module doc as well? In fact I think most of your `SigmaCompleteLattice` docstring could be moved to the module doc